nms/app/packages/magmalte/app/components/lte/LteSections.js

```diff
@@ -119,33 +254,74 @@ export function getLteSections(
   return sections;
 }
 
-export function getLteSectionsV2(alertsEnabled: boolean): SectionsConfigs {
+export function getLteSectionsV2(
+  alertsEnabled: boolean,
+  lteContext: ?LteContextType,
+): SectionsConfigs {
   const sections = [
     'dashboard', // landing path
     [
       {
         path: 'dashboard',
         label: 'Dashboard',
         icon: <DashboardIcon />,
-        component: LteDashboard,
+        component: lteContext
+          ? () => (
+              <EnodebContext.Provider value={lteContext.enodebCtx}>
+                <GatewayContext.Provider value={lteContext.gatewayCtx}>
+                  <LteDashboard />
+                </GatewayContext.Provider>
+              </EnodebContext.Provider>
+            )
+          : LoadingFiller,
       },
       {
         path: 'equipment',
         label: 'Equipment',
         icon: <RouterIcon />,
-        component: EquipmentDashboard,
+        component: lteContext
+          ? () => (
+              <EnodebContext.Provider value={lteContext.enodebCtx}>
```

Comment:
I really don't know much about these Context.Providers, but is there a way we can avoid all the indentation? It's already quite a bit and I'm worried about a pattern of just continually increasing the depth of the <EquipmentDashboard> every time we add some context

Reply:
I feel like this is probably just the way to do these things, which is totally fine then.

Reply:
Yeah. I'm not sure how I can remove this level of indentation. I could perhaps wrap it into a function like the AppContextProvider method
```export function AppContextProvider(props: Props) {
  const config: {appData: AppContextAppData} = window.CONFIG;
  const {appData} = config;
  const value = {
    ...appContextDefaults,
    ...appData,
    networkIds: props.networkIDs || [],
    isTabEnabled: (tab: Tab): boolean => {
      return appData.tabs?.indexOf(tab) !== -1;
    },
    isFeatureEnabled: (featureID: FeatureID): boolean => {
      return appData.enabledFeatures.indexOf(featureID) !== -1;
    },
  };

  return (
    <AppContext.Provider value={value}>{props.children}</AppContext.Provider>
  );
}```

However since I probably use different context data in different components, might not be very useful.